### PR TITLE
Add ws_queue type

### PR DESCRIPTION
**Pull request by matthiasbeyer**
Targets #597.

~~Pair-programmed-with: Marcel Müller com2040@gmail.com~~
Pair-programmed-with: @TheNeikos

### CODE REVIEW

**Comment by neithernut** [comment]
_Has_ it to be an object (as in `ws_object`)?

**Comment by matthiasbeyer** [comment]
If you mean the `ws_queue` type: It gives us easy locking and the like.

If you mean the values inside: I assumed that `ws_object`s are of relevance here, as it is written to hold events, if I'm right...

**Comment by neithernut** [comment]
I mean the queue itself.
I do not think that it should be a `ws_object` _just_ because it automatically introduces locking facilities.

**Comment by TheNeikos** [comment]
There is no reason for me for it to _not_ be a `ws_object`. It makes it easier to use and it fits our setup.

File: src/objects/queue.c [hunk 1]
```diff
@@ -0,0 +1,168 @@
...
+
+struct ws_object*
+ws_queue_pop(
+    struct ws_queue* q
+) {
+    if (!q || wl_list_empty(&q->link)) {
+        return NULL;
+    }
+
+    ws_object_lock_write(&q->obj);
```
Comment (neithernut) [inline src/objects/queue.c]: You forgot to check the return value of this call.

**Comment by neithernut** [comment]
ACK for squashing.

**Comment by neithernut** [comment]
ACK.

**Pull request merged**